src/ambianic/pipeline/ai/fall_detect.py

```diff
@@ -171,34 +185,41 @@ def fall_detect(self, image=None):
                 if not self._prev_vals or lapse > self.max_time_between_frames:
                     log.info("No recent pose to compare to. Will save this frame pose for subsequent comparison.")
                 else:
-                    left_angle = 0
-                    left_body_line_detected = len(pose_vals_list[0]) == 2
-                    if left_body_line_detected and (self._prev_vals[0] is not None):
-                        try:
-                            temp_left_point = [[self._prev_vals[0][0], self._prev_vals[0][1]], [pose_vals_list[0][0], pose_vals_list[0][1]]]
-                            left_angle = self.calculate_angle(temp_left_point)
-                            log.info("Left shoulder-hip angle: %r", left_angle)
-                        except IndexError as e:
-                            log.exception("Error while calculating angle: %r", e)
-                    right_angle = 0
-                    right_body_line_detected = len(pose_vals_list[1]) == 2
-                    if right_body_line_detected and (self._prev_vals[1] is not None):
-                        try:
-                            temp_right_point = [[self._prev_vals[1][0], self._prev_vals[1][1]], [pose_vals_list[1][0], pose_vals_list[1][1]]]
-                            right_angle = self.calculate_angle(temp_right_point)
-                            log.info("Right shoulder-hip angle: %r", left_angle)
-                        except IndexError as e:
-                            log.exception("Error while calculating angle: %r", e)
-
-                    angle_change = max(left_angle, right_angle)
-                    if angle_change > self._fall_factor:
-                        # insert a box that covers the whole image as a workaround
-                        # to meet the expected format of the save_detections element
-                        box = [0, 0, 1, 1]
-                        inference_result.append(('FALL', pose.score, box, angle_change ))
-                        log.info("Fall detected: %r", inference_result)
+                    if (left_angle_with_yaxis and self._prev_left_angle_with_yaxis and left_angle_with_yaxis > self._prev_left_angle_with_yaxis) or (rigth_angle_with_yaxis and self._prev_right_angle_with_yaxis and rigth_angle_with_yaxis > self._prev_right_angle_with_yaxis):
+                        left_angle = 0
```

Comment:
@bhavikapanara I think your new code for calculating body line angle with y-axis needs to be guarded with these checks. Otherwise we run the risk of exceptions in cases when we don't have a full body line detected, but only a single point (hip only or shoulder only).

Reply:
@ivelin updated the code